backend/src/User.ts

```diff
@@ -0,0 +1,66 @@
+import { UserDBEntity } from './entities/UserDBEntity';
+import * as bcrypt from 'bcrypt';
+
+export enum AUTH_RESULT_CODE {
+  SUCCESS,
+  UNKNOWN_EMAIL,
+  BAD_PASSWORD,
+}
+
+export enum REG_RESULT_CODE {
+  SUCCESS,
+  NON_UNIQ_EMAIL,
+  NON_UNIQ_USERNAME,
+  WEAK_PASSWORD,
+}
+
+const SALT_ROUNDS = 10;
+
+export default class User {
```

Comment:
This class is weird to me... What does it represent, a user? But then you only store the username, nothing else... So it is more like the username ? with some database functions?

Honestly, I think a "User" should be something like this:

```
interface User {
  username: string;
  password?: string;
  email?: string;
}
```

And these methods are only "helper functions" that use "User"